src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -390,4 +479,35 @@ internal void InvalidateCache()
         }
     }
 
+    private void AddCachedExtensionToExtensionMap<T>(
+        IDictionary<string, ISet<string>> extensionMap,
+        string extensionType,
+        IList<T> extensions) where T : TestPluginInformation
+    {
+        if (extensions == null) return;
+
+        extensionMap.Add(extensionType, new HashSet<string>());
+        foreach (var extension in extensions)
+        {
+            extensionMap[extensionType].Add(extension.IdentifierData);
+        }
+    }
+
+    private static string SerializeExtensionMap(IDictionary<string, ISet<string>> extensionMap)
+    {
+        StringBuilder sb = new();
+
+        foreach (var kvp in extensionMap)
+        {
+            sb.Append(
+                kvp.Value?.Count == 0
+                ? string.Empty
+                : string.Format(
+                    "{0}=[{1}];",
+                    kvp.Key,
+                    string.Join(",", kvp.Value.ToArray())));
```

Comment:
You don't need the `ToArray` call as there is an `IEnumerable` overload.

```suggestion
                    string.Join(",", kvp.Value)));
```

Reply:
Fixed, thanks.